waiter/integration/waiter/async_request_integration_test.clj

```diff
@@ -187,21 +187,20 @@
           request-processing-time-ms 60000
           num-threads 20
           num-requests-to-delete 10
-          status-locations
-          (parallelize-requests
-            num-threads 1
-            (fn []
-              (log/info "making kitchen request")
-              (let [async-request-headers
-                    (assoc request-headers :x-kitchen-delay-ms (int (* (max 0.5 (double (rand))) request-processing-time-ms))
-                                           :x-kitchen-store-async-response-ms 600000)
-                    {:keys [headers] :as response}
-                    (make-kitchen-request waiter-url async-request-headers :http-method-fn http/get :body "" :path "/async/request")
-                    status-location (get (pc/map-keys str/lower-case headers) "location")]
-                (assert-response-status response 202)
-                (is (not (str/blank? status-location)))
-                status-location))
-            :verbose true)]
+          async-responses (parallelize-requests
+                            num-threads 1
+                            (fn []
+                              (log/info "making kitchen request")
+                              (let [async-request-headers
+                                    (assoc request-headers :x-kitchen-delay-ms (int (* (max 0.5 (double (rand))) request-processing-time-ms))
+                                                           :x-kitchen-store-async-response-ms 600000)]
+                                (make-kitchen-request waiter-url async-request-headers :http-method-fn http/get :body "" :path "/async/request")))
+                            :verbose true)
+          status-locations (for [{:keys [headers] :as response} async-responses]
+                             (let [status-location (get (pc/map-keys str/lower-case headers) "location")]
```

Comment:
`clj-http` does not, e.g. we have other tests where we check headers such as `X-Waiter-Router-Id`